corelibs/libcache.md

```diff
@@ -0,0 +1,187 @@
+|--------|-------------|--------------| ------- |
+| Implemented <br /><br /> Tests in progress| Reads into a `buffer` up to `count` bytes from a cache line mapped from address `addr`. | **On success:** a number of bytes read from cache <br /><br /> **On failure:** an error number | Fails if `buffer` is NULL or/and `addr` goes beyond the scope of the cached source memory. (`EINVAL`).  <br /><br /> May fail due to problems with the cached source memory (`EIO`). <br /><br /> The behavior is undefined if read occurs beyond the end of the `buffer`.
+
+```c
+ssize_t cache_write(cachectx_t *cache, uint64_t addr, void *buffer, size_t count, int policy);
+```
+| Status | Description | Return value | Remarks |
+|--------|-------------|--------------| ------- |
+| Implemented <br /><br />Tests in progress | Writes to the cache up to a `count` of bytes from a `buffer` according to `policy` (see: [Write policy](#write-policy)) by creating new and/or updating existing cache lines. Sets the validity bit for new lines. <br /><br /> | **On success:** a number of bytes written to the cache <br /><br /> **On failure:** an error number | Fails if `buffer` is NULL or/and wrong `policy` value is supplied (`EINVAL`). Fails when `addr` goes beyond the scope of the cached source memory. <br /><br /> May fail due to problems with cached the source memory (`EIO`). <br /><br /> The behavior is undefinded if write occurs beyond the end of the `buffer`.
+
+```c
+int cache_flush(cachectx_t *cache, const uint64_t begAddr, const uint64_t endAddr);
+```
+| Status | Description | Return value | Remarks |
+|--------|-------------|--------------| ------- |
+| Implemented <br /><br />Tests in progress | Flushes a range of cache lines starting from an address `begAddr` up to `endAddr`. <br /><br />  Writes dirty lines (see: [Write policy](#write-policy)) marked with dirty bit to the cached source memory. Clears the dirty bit. | **On success:** 0 (i.e. all bytes from all lines marked with the dirty bit in given range were successfully written to the cached source memory) <br /><br /> **On failure:** an error number | Fails if `begAddr` is greater than `endAddr` or/and `begAddr` is greater than `srcMemSize` (`EINVAL`). <br /><br /> May fail due to problems with the cached source memory (`EIO`).
+
+```c
+int cache_invalidate(cachectx_t *cache, const uint64_t begAddr, const uint64_t endAddr);
+```
+| Status | Description | Return value | Remarks |
+|--------|-------------|--------------| ------- |
+| Implemented <br /><br />Tests in progress | Invalidates a range of cache lines starting from an address `begAddr` up to `endAddr`. <br /><br /> Clears the validity bit for lines in that range. | **On success:** 0 (i.e. all lines marked with the validity bit in the given range were successfully invalidated) <br /><br /> **On failure:** an error number  | Fails if `begAddr` is greater than `endAddr` or/and `begAddr` is greater than `srcMemSize` (`EINVAL`).
```

Comment:
I would be good to mention that dirty lines are not flushed by this operation, this is quite important detail ;)